contributors/devel/kubectl-conventions.md

```diff
@@ -375,6 +376,21 @@ this is not a mandatory structure and not every command is implemented this way,
 but this is a nice convention so try to be compliant with it. As an example,
 have a look at how [kubectl logs](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/logs.go) is implemented.
 
+## Exit code conventions
+
+Generally, for all the commands exit code, result of `zero` means succeed and `non-zero` means error.
+
+For some declarative commands like `apply` may have a situation which command exited success but without actual changes, we provide a flag `--error-unchanged` to specify the exit code to 2 if necessary.
+
+For other commands like `create` that return error when unchanged, we have the flag `--ignore-unchanged` to make it return 0 and no errors.
+
+
+| Exit Code Number | Meaning | Enable |
+| :---             | :---    | :---    |
+| 0                | Command exited success  | By default, By flag `--ignore-unchanged` |
+| 1                | Command exited for general errors  | By default |
+| 2                | Command exited success but without changes | By flag `--error-unchanged`|
```

Comment:
Probably rephrase this to `Command was successful, but the user requested a distinct exit code when no change was made`.

Also, we shouldn't be using 2 which is reserved as a bash shell error.  We should use 3 or higher.  See http://www.tldp.org/LDP/abs/html/exitcodes.html